### PR TITLE
calendar and world calendar enhancement

### PR DESCRIPTION
**Pull request by aminbksi**
https://momentum.nifty.pm/l/Lm0E7vKmrYW

moving eventList and eventForm (Stores) in core/model
moving DeleteEventDialog to ui-kit
added edit form for world calendar

### CODE REVIEW

File: src/core/models/EventItem/EventItem.ts [hunk 1]
```diff
@@ -70,51 +61,54 @@ const EventItemModel = types
...
-        description: self.description,
-        duration: durationInHours(self.start, self.end),
-        endDatetime: formattedStringFromDate(self.end),
-        startDatetime: formattedStringFromDate(self.start),
-        title: self.title,
+        description: self.data?.description ?? '',
+        duration: durationInHours(self.data?.start ?? new Date(), self.data?.end ?? new Date()),
+        endDatetime: formattedStringFromDate(self.data?.end ?? new Date()),
+        startDatetime: formattedStringFromDate(self.data?.start ?? new Date()),
+        title: self.data?.title ?? '',
```
Comment (dmitry-yudakov) [inline src/core/models/EventItem/EventItem.ts:111]: maybe `return null` or undefined if `!self.data`?

File: src/scenes/worldCalendar/pages/WorldCalendarPage/components/templates/EventForm/EventForm.tsx [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+          />
+        </styled.Item>
+        <styled.FileUploaderItem>
+          <styled.TileImageUpload>
+            {(image || currentEvent?.image_hash) && (
+              <styled.ImagePreview
+                src={
+                  (image && URL.createObjectURL(image)) ||
+                  (currentEvent?.image_hash &&
+                    `${appVariables.RENDER_SERVICE_URL}/get/${currentEvent?.image_hash}`) ||
```
Comment (PAKovalev) [inline src/scenes/worldCalendar/pages/WorldCalendarPage/components/templates/EventForm/EventForm.tsx:231]: Move to the model

File: src/core/models/EventItem/EventItem.ts [hunk 3]
```diff
@@ -70,51 +61,54 @@ const EventItemModel = types
     },
     isLive(): boolean {
       const nowDate = new Date();
-      return nowDate >= self.start && nowDate <= self.end;
+      if (self.data) {
+        return nowDate >= self.data?.start && nowDate <= self.data?.end;
```
Comment (PAKovalev) [inline src/core/models/EventItem/EventItem.ts:65]: remove `?`

File: src/scenes/collaboration/stores/CalendarStore/CalendarStore.ts [hunk 4]
```diff
@@ -3,25 +3,25 @@ import {cast, flow, types} from 'mobx-state-tree';
...
 const CalendarStore = types.compose(
   ResetModel,
   types
     .model('CalendarStore', {
       formDialog: types.optional(DialogModel, {}),
       magicDialog: types.optional(DialogModel, {}),
       deleteConfirmationDialog: types.optional(DialogModel, {}),
-      eventListStore: types.optional(EventListStore, {}),
-      eventFormStore: types.optional(EventFormStore, {}),
+      eventListStore: types.optional(EventList, {}),
```
Comment (PAKovalev) [inline src/scenes/collaboration/stores/CalendarStore/CalendarStore.ts:23]: `eventList`

File: src/ui-kit/organisms/EventItem/EventItem.tsx [hunk 5]
```diff
@@ -55,14 +58,14 @@ const EventItem: FC<PropsInterface> = ({
   const buttons = () => (
     <styled.Buttons className="base">
       <styled.Buttons>
-        {isWorldCalendar && event.spaceId && (
+        {isWorldCalendar && event.data?.spaceId && (
           <styled.EventButton
             label={`${t('eventList.eventItem.flyToSpace')} ${
-              event.spaceName && event.spaceName.slice(0, 12)
-            } ${event.spaceName && (event.spaceName.length > 12 ? '...' : '')}`}
+              event.data?.spaceName && event.data?.spaceName.slice(0, 12)
```
Comment (dmitry-yudakov) [inline src/ui-kit/organisms/EventItem/EventItem.tsx:64]: maybe just `event.data?.spaceName?.slice(0, 12)`? No need to check for `event.data?.spaceName` separately as I see.

File: src/scenes/worldCalendar/pages/WorldCalendarPage/components/templates/EventForm/EventForm.styled.ts [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+  padding-top: 20px;
+`;
+
+export const FileUploaderItem = styled.div`
+  flex: 50%;
+  padding-top: 25px;
+`;
+
+export const DateInput = styled.div`
+  .react-datepicker {
```
Comment (PAKovalev) [inline src/scenes/worldCalendar/pages/WorldCalendarPage/components/templates/EventForm/EventForm.styled.ts:49]: Move to Global styles

File: src/scenes/worldCalendar/stores/WorldCalendarStore/WorldCalendarStore.ts [hunk 7]
```diff
@@ -1,21 +1,60 @@
...
   types
     .model('WorldCalendarStore', {
+      formDialog: types.optional(DialogModel, {}),
+      deleteConfirmationDialog: types.optional(DialogModel, {}),
+      removeEventRequest: types.optional(RequestModel, {}),
+      eventIdToRemove: types.maybe(types.string),
       weblinkDialog: types.optional(DialogModel, {}),
       magicDialog: types.optional(DialogModel, {}),
-      eventListStore: types.optional(EventListStore, {}),
+      eventListStore: types.optional(EventList, {}),
```
Comment (PAKovalev) [inline src/scenes/worldCalendar/stores/WorldCalendarStore/WorldCalendarStore.ts:24]: eventList

File: src/ui-kit/organisms/EventItem/EventItem.tsx (hunk 5 above)
Comment (dmitry-yudakov) [reply src/ui-kit/organisms/EventItem/EventItem.tsx:64]: And perhaps move this whole formatting logic to separate function?

File: src/ui-kit/organisms/EventItem/EventItem.tsx [hunk 8]
```diff
@@ -72,21 +75,21 @@ const EventItem: FC<PropsInterface> = ({
...
+        {!!event.data?.web_link && (
           <styled.EventButton
             label={t('eventList.eventItem.websiteLink')}
             icon="link"
             transform="capitalized"
             onClick={() => {
-              if (event?.web_link) {
-                onWeblinkClick(event.web_link);
+              if (event.data?.web_link) {
+                onWeblinkClick(event.data?.web_link);
```
Comment (dmitry-yudakov) [inline src/ui-kit/organisms/EventItem/EventItem.tsx:92]: You already check `event.data?.web_link` at the top - probably no need check in the handler as well?

File: src/ui-kit/organisms/EventItem/EventItem.tsx [hunk 9]
```diff
@@ -4,25 +4,27 @@ import AddToCalendarHOC, {SHARE_SITES} from 'react-add-to-calendar-hoc';
...
+  onEdit?: (event: EventItemInterface) => void;
+  onRemove?: (event: EventItemInterface) => void;
   onMagicLinkOpen: (eventId: string, spaceId?: string) => void;
   zIndex?: number;
   isWorldCalendar?: boolean;
   onFlyToGathering?: (spaceId: string) => void;
   onFlyToSpace?: (spaceId: string) => void;
   onWeblinkClick: (weblink: string) => void;
   onShowAttendeesList: (eventName: string, eventId: string, spaceId: string) => void;
+  isSpace?: boolean;
```
Comment (PAKovalev) [inline src/ui-kit/organisms/EventItem/EventItem.tsx:26]: canManageInSpace
canManageInWorld

File: src/ui-kit/organisms/EventItem/components/Actions/Actions.tsx [hunk 10]
```diff
@@ -1,37 +1,33 @@
...
+          onRemove?.(event);
+        }}
+      />
+      <SvgButton
+        iconName="edit"
+        size="medium"
+        onClick={() => {
+          onEdit?.(event);
+        }}
+      />
```
Comment (dmitry-yudakov) [inline src/ui-kit/organisms/EventItem/components/Actions/Actions.tsx:30]: If no callbacks passed we're showing buttons and just ignoring the clicks? Maybe change the interface and make onRemove/onEdit mandatory? Or leave the `{!!onEdit &&` checks?

**Review by dmitry-yudakov — APPROVED** [review]

File: src/scenes/worldCalendar/pages/WorldCalendarPage/components/templates/EventForm/EventForm.tsx [hunk 11]
```diff
@@ -0,0 +1,240 @@
...
+    setError,
+    clearErrors
+  } = useForm<EventFormInterface>();
+  const [image, setImage] = useState<File>();
+  const [startDate, setStartDate] = useState<Date>(currentEvent?.start ?? new Date());
+
+  const [endDate, setEndDate] = useState<Date>(currentEvent?.end ?? new Date());
+
+  const formSubmitHandler: SubmitHandler<EventFormInterface> = async (data: EventFormInterface) => {
+    if (data.web_link?.length === 0) {
```
Comment (PAKovalev) [inline src/scenes/worldCalendar/pages/WorldCalendarPage/components/templates/EventForm/EventForm.tsx:37]: `!data.web_link?.length`

**Review by PAKovalev — APPROVED** [review]

**Pull request merged**